src/test/java/com/team4/testingsystem/controllers/CoachGradeControllerIntegrationTest.java

```diff
@@ -0,0 +1,274 @@
+public class CoachGradeControllerIntegrationTest {
+    private final MockMvc mockMvc;
+
+    private final UsersRepository usersRepository;
+    private final QuestionRepository questionRepository;
+    private final TestsRepository testsRepository;
+    private final CoachGradeRepository gradeRepository;
+
+    private final ObjectMapper objectMapper;
+
+    private User user;
+    private CustomUserDetails userDetails;
+
+    @Autowired
+    public CoachGradeControllerIntegrationTest(MockMvc mockMvc,
+                                               UsersRepository usersRepository,
+                                               QuestionRepository questionRepository,
+                                               TestsRepository testsRepository,
+                                               CoachGradeRepository gradeRepository,
+                                               ObjectMapper objectMapper) {
+        this.mockMvc = mockMvc;
+        this.usersRepository = usersRepository;
+        this.questionRepository = questionRepository;
+        this.testsRepository = testsRepository;
+        this.gradeRepository = gradeRepository;
+        this.objectMapper = objectMapper;
+    }
+
+    @BeforeEach
+    void init() {
+        user = usersRepository.findByLogin("rus_user@northsixty.com").orElseThrow();
+        userDetails = new CustomUserDetails(user);
+    }
+
+    @Test
+    void getGrade() throws Exception {
+        mockMvc.perform(get("/grades/100/100")
+                .with(user(userDetails)))
+                .andExpect(status().isNotFound());
+
+        com.team4.testingsystem.entities.Test test = EntityCreatorUtil.createTest(user);
+        testsRepository.save(test);
+
+        mockMvc.perform(get("/grades/{testId}/100", test.getId())
+                .with(user(userDetails)))
+                .andExpect(status().isNotFound());
+
+        Question question = EntityCreatorUtil.createQuestion(user);
+        questionRepository.save(question);
+
+        mockMvc.perform(get("/grades/{testId}/{questionId}", test.getId(), question.getId())
+                .with(user(userDetails)))
+                .andExpect(status().isNotFound());
+
+        CoachGrade grade = new CoachGrade(test, question, 8);
+        gradeRepository.save(grade);
+
+        MvcResult mvcResult = mockMvc.perform(get("/grades/{testId}/{questionId}", test.getId(), question.getId())
+                .with(user(userDetails)))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        String response = mvcResult.getResponse().getContentAsString();
+        CoachGradeDTO gradeDTO = objectMapper.readValue(response, CoachGradeDTO.class);
+
+        Assertions.assertEquals(new CoachGradeDTO(grade), gradeDTO);
+
+        gradeRepository.delete(grade);
+        questionRepository.delete(question);
+        testsRepository.delete(test);
+    }
+
+    @Test
+    void getGrades() throws Exception {
+        mockMvc.perform(get("/grades/100")
```

Comment:
Are you sure it will not be created by a previous test?

Reply:
All tests delete entities they create

Reply:
If it passed successfully...

Reply:
I can clear tests, questions and grades table before each test

Reply:
Why can't we just use another id?

Reply:
Sequence that generates IDs doesn't restart, so I think it will generate different IDs. So, correctness will depend on the order of the tests?
In my opinion, clearing tables is more correct)

Reply:
Both, parallel run and different order is possible